pkgs/sage-setup/pyproject.toml

```diff
@@ -0,0 +1,47 @@
+[build-system]
+requires = ["setuptools>=61.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "sage-setup"
+description = "Sage: Open Source Mathematics Software: Build system of the Sage library"
+license = {text = "GNU General Public License (GPL) v2 or later"}
+authors = [{name = "The Sage Developers", email = "sage-support@googlegroups.com"}]
+classifiers = [
+    "Development Status :: 6 - Mature",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)",
+    "Operating System :: POSIX",
+    "Operating System :: MacOS :: MacOS X",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering :: Mathematics",
+]
+urls = {Homepage = "https://www.sagemath.org"}
+requires-python = ">=3.9, <3.12"
+dependencies = [
+    "pkgconfig",
+    "jinja2",
```

Comment:
What about cython? I believe both jinja2 and cython are runtime dependencies.

Reply:
The distributions that use Cython declare that in their build-system requires. I don't think we need it here.

Reply:
That's an argument for distribution to declare their use of jinja2 as well. And there is actually no call to pkgconfig in sage_setup.

Reply:
Probably jinja2 should actually be an optional dependency (formerly known as extras-require) because only the packages that you sage_setup.autogen use it.

Reply:
Good point about pkgconfig.

Reply:
> Probably jinja2 should actually be an optional dependency (formerly known as extras-require) because only the packages that you sage_setup.autogen use it.

Yes, and only called from one key distribution right now, I guess it is fair.

Reply:
After modularization, there will be more uses, but I've made the change now.